app/src/main/java/org/mozilla/fenix/home/recentvisits/RecentlyVisitedItem.kt

```diff
@@ -21,18 +18,4 @@ sealed class RecentlyVisitedItem {
         val title: String,
         val url: String
     ) : RecentlyVisitedItem()
-
-    /**
-     * A group of previously accessed webpages related by their search terms.
-     *
-     * @property title The title of the group.
-     * @property historyMetadata A list of [HistoryMetadata] records that matches the title.
-     */
-    data class RecentHistoryGroup(
-        val title: String,
-        val historyMetadata: List<HistoryMetadata> = emptyList()
-    ) : RecentlyVisitedItem()
```

Comment:
For sure. I wasn't sure if that would be too big a lift for the purposes of this PR, since just `RecentlyVisitedItem.RecentHistoryHighlight` has something like 54 usages.

Reply:
Ah, 54 is pretty high. I'll defer to whatever you choose to do there since this PR is pretty large already